integration-tests/tests/src/tests/sync/mixed.ts

```diff
@@ -16,28 +16,66 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 import { expect } from "chai";
-import Realm from "realm";
+import Realm, { Mixed, ObjectSchema } from "realm";
 
 import { importAppBefore, authenticateUserBefore, openRealmBefore } from "../../hooks";
 
 import { itUploadsDeletesAndDownloads } from "./upload-delete-download";
 import { buildAppConfig } from "../../utils/build-app-config";
 
-type MixedClass = {
-  _id: Realm.BSON.ObjectId;
-  value: Realm.Mixed;
-  list: Realm.List<Realm.Mixed>;
-};
 type Value = Realm.Mixed | ((realm: Realm) => Realm.Mixed);
-type ValueTester = (actual: Realm.Mixed, inserted: Realm.Mixed) => void | boolean;
+type ValueTester = (actual: Realm.Mixed, inserted: Realm.Mixed) => void;
+
+class MixedClass extends Realm.Object<MixedClass> {
+  _id!: Realm.BSON.ObjectId;
+  value: Realm.Mixed;
+  list!: Realm.List<Realm.Mixed>;
+  dict!: Realm.Dictionary<Realm.Mixed>;
+
+  static schema: ObjectSchema = {
+    name: "MixedClass",
+    properties: {
+      _id: "objectId",
+      value: "mixed",
+      list: "mixed[]",
+      dict: "mixed{}",
+    },
+    primaryKey: "_id",
+  };
+}
 
 /**
  * The default tester of values.
  * @param actual The value downloaded from the server.
  * @param inserted The value inserted locally before upload.
  */
 function defaultTester(actual: Realm.Mixed, inserted: Realm.Mixed) {
-  expect(actual).equals(inserted);
+  if (actual instanceof Realm.List) {
+    const insertedVal = inserted as Realm.Mixed[]; //TODO I should remove all of these "as" (?)
```

Comment:
Or should the type for `inserted` be `any` instead of `Realm.Mixed`?

Reply:
That could also work. Not sure what is the preferred way for this here

Reply:
Never `any` ... `unknown` at best.